### PR TITLE
Add GeoGebra 5

### PR DESCRIPTION
**Pull request by tech189**
Adding the old/classic version of GeoGebra. Closes request #275

### CODE REVIEW

**Comment by rashil2000** [comment]
Don't use `cd`, it can cause unexpected results when the installation ends.
Use absolute path names instead.

**Review by rashil2000 — APPROVED** [review]

File: bucket/geogebra5.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{
+    "version": "5.0.675.0",
+    "description": "A dynamic mathematics software for education that brings together geometry, algebra, spreadsheets, graphing, statistics and calculus.",
+    "homepage": "https://www.geogebra.org/",
+    "license": {
+        "identifier": "freeware for non-commercial use",
```
Comment (issaclin32) [inline bucket/geogebra5.json:6]:
```suggestion
        "identifier": "Freeware",
```

File: bucket/geogebra5.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    ],
+    "shortcuts": [
+        [
+            "GeoGebra.exe",
+            "GeoGebra 5"
+        ]
+    ],
+    "checkver": {
+        "url": "https://download.geogebra.org/installers/5.0/version.txt",
+        "re": "(\\d+)-(\\d+)-(\\d+)-(\\d+)",
```
Comment (issaclin32) [inline bucket/geogebra5.json:32]:
```suggestion
        "regex": "(\\d+)-(\\d+)-(\\d+)-(\\d+)",
```
we use `regex` now (in all official packages and manuals), we keep `re` functional for backward compatibility.

File: bucket/geogebra5.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        "replace": "${1}.${2}.${3}.${4}"
+    },
+    "autoupdate": {
+        "url": "https://download.geogebra.org/installers/5.0/GeoGebra-Windows-Portable-$dashVersion.zip"
```
Comment (issaclin32) [inline bucket/geogebra5.json:36]:
```suggestion
        "url": "https://download.geogebra.org/installers/5.0/GeoGebra-Windows-Portable-$dashVersion.zip",
        "extract_dir": "geogebra $version"
```
Use `extract_dir` instead. This is clearer, and easier to maintain.

File: bucket/geogebra5.json [hunk 4]
```diff
@@ -0,0 +1,38 @@
+{
+    "version": "5.0.675.0",
+    "description": "A dynamic mathematics software for education that brings together geometry, algebra, spreadsheets, graphing, statistics and calculus.",
+    "homepage": "https://www.geogebra.org/",
+    "license": {
+        "identifier": "freeware for non-commercial use",
+        "url": "https://www.geogebra.org/license"
+    },
+    "url": "https://download.geogebra.org/installers/5.0/GeoGebra-Windows-Portable-5-0-675-0.zip",
+    "hash": "0d6b8d21ee44e68ddd00337fbf7672af4318ad9fd3f1b88cd5230fd5f79d1163",
```
Comment (issaclin32) [inline bucket/geogebra5.json:10]:
```suggestion
    "hash": "0d6b8d21ee44e68ddd00337fbf7672af4318ad9fd3f1b88cd5230fd5f79d1163",
    "extract_dir": "geogebra 5.0.675.0",
```

File: bucket/geogebra5.json [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    "installer": {
```
Comment (issaclin32) [inline bucket/geogebra5.json:11]: Remove `installer`, because it can be replaced by `extract_dir`.

**Comment by issaclin32** [comment]
By the way, it is a 32-bit app. So `architecture` is not needed 
(no need to modify anything, this is just a reminder)
```
> pelook $Env:Scoop\apps\geogebra5\current\geogebra.exe
loaded "D:\scoop\apps\geogebra5\current\geogebra.exe" / 196880 (0x30110) bytes
signature/type:       PE32 EXE image for i386
image checksum:       0x00031E1C (OK)
machine:              0x014C (i386)
subsystem:            2 (Windows GUI)
minimum os:           4.0 (Win95/NT4)
linkver:              2.34
```

File: bucket/geogebra5.json (hunk 5 above)
Comment (tech189) [reply bucket/geogebra5.json:11]: See conversation below, it's not possible sadly.

File: bucket/geogebra5.json (hunk 4 above)
Comment (tech189) [reply bucket/geogebra5.json:10]: Keeping the directory hardcoded in the manifest would be hard to maintain. See discussion below.

File: bucket/geogebra5.json (hunk 3 above)
Comment (issaclin32) [reply bucket/geogebra5.json:36]: Oops. Should be `geogebra 5.0.675.0`. I didn't put it correctly.
Autoupdate for `extract_dir` will not work if `extract_dir` does not exist in the manifest.

Comment (tech189) [reply bucket/geogebra5.json]: No worries, you're totally right. Now it's much simpler without the installer script! :D

**Review by issaclin32 — APPROVED** [review]

**Comment by tech189** [comment]
Awesome, thanks everyone! I'm thinking of also making a PR for the current version of GeoGebra (v6), would that fit best in the `extras` bucket?

**Comment by issaclin32** [comment]
> Awesome, thanks everyone! I'm thinking of also making a PR for the current version of GeoGebra (v6), would that fit best in the `extras` bucket?

You are right. I will be grateful if you could send a PR.

**Pull request merged**